lib/widgets/profile.dart

```diff
@@ -0,0 +1,261 @@
+      final widget = _renderCustomProfileFieldValue(context, profileField.value, realmField);
+      if (widget == null) continue;
+      entries.add((label: realmField.name, widget: widget));
+    }
+
+    return DefaultTextStyle.merge(
+      style: const TextStyle(fontSize: 15, height: 1),
+      textAlign: TextAlign.left,
+      child: Column(
+        children: [
+          const SizedBox(height: 16),
+          Table(
+            border: TableBorder.all(style: BorderStyle.none),
+            columnWidths: const {
+              0: FixedColumnWidth(96 + 8),
+              1: FlexColumnWidth()},
+            defaultVerticalAlignment: TableCellVerticalAlignment.baseline,
+            textBaseline: TextBaseline.alphabetic,
+            children: entries.map((entry) => TableRow(
+              children: [
+                Padding(
+                  padding: const EdgeInsets.fromLTRB(0, 4, 8, 4),
+                  child: Text(entry.label, style: const TextStyle(fontWeight: FontWeight.bold))),
+                entry.widget,
+              ])).toList()),
+        ]));
+  }
+
+  Widget? _renderCustomProfileFieldValue(BuildContext context, String value, CustomProfileField realmField) {
+    switch (realmField.type) {
+      case CustomProfileFieldType.link:
+        return _buildLinkWidget(context, value, value);
+      case CustomProfileFieldType.choice:
+        return _buildChoiceField(context, value, realmField);
+      case CustomProfileFieldType.externalAccount:
+        return _buildExternalAccountField(context, value, realmField);
+      case CustomProfileFieldType.user:
+        return _buildUserField(context, value, realmField);
+      case CustomProfileFieldType.date:
+        // TODO(server): The value's format is undocumented, but empirically
+        //   it's a date in ISO format, like 2000-01-01.
+        // That's readable as is, but:
+        // TODO format this date using user's locale.
+        return _buildTextWidget(value);
+      case CustomProfileFieldType.shortText:
+      case CustomProfileFieldType.longText:
+      case CustomProfileFieldType.pronouns:
+        return _buildTextWidget(value);
+      case CustomProfileFieldType.unknown:
+        return null;
+    }
+  }
+
+  Widget? _buildChoiceField(BuildContext context, String value, CustomProfileField realmField) {
+    // TODO(server): This isn't really documented.  But see chat thread:
+    //   https://chat.zulip.org/#narrow/stream/378-api-design/topic/custom.20profile.20fields/near/1383005
+    final CustomProfileFieldChoices choiceFieldData;
+    try {
+      final realmFieldData = jsonDecode(realmField.fieldData);
+      choiceFieldData = CustomProfileFieldChoices.fromJson(realmFieldData);
+    } on FormatException {
+      return null;
+    } on TypeError {
+      return null;
+    }
+    final choiceItem = choiceFieldData.choices[value];
+    if (choiceItem == null) return null;
+    return _buildTextWidget(choiceItem.text);
```

Comment:
Together with the `if (choiceFieldData == null) return null;` line in my suggestion above, this can be compacted to:
```suggestion
    final choiceText = choiceFieldData?.choices[value]?.text;
    return choiceText == null ? null : _buildTextWidget(choiceText);
```